akka-statsd-http-client/src/main/scala/StatsClient.scala

```diff
@@ -0,0 +1,100 @@
+package akka.statsd.http.client
+
+import akka.actor.{ActorRef, ActorSystem}
+import akka.http.scaladsl.HttpExt
+import akka.statsd._
+import akka.http.scaladsl.model._
+import akka.statsd.{Config => StatsConfig}
+import scala.concurrent.{ExecutionContext, Future}
+
+class StatsClient(
+  requestMaker: HttpRequest => Future[HttpResponse],
+  baseBucket: String = "http.client"
+)(implicit system: ActorSystem) {
+
+  /**
+    * Override this in the calling class when you want to specify your own configuration
+    */
+  def statsConfig: StatsConfig = StatsConfig()
+
+  protected def nowInMillis: Long = System.currentTimeMillis
```

Comment:
nitpicking, but I think it's safer to use nanoTime in this case because it's at least guaranteed to always go forward and not jump around (due to NTP syncs for example).